configsuite/extension/ext.py

```diff
@@ -0,0 +1,206 @@
+"""Copyright 2018 Equinor ASA and The Netherlands Organisation for
+Applied Scientific Research TNO.
+
+Licensed under the MIT license.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the conditions stated in the LICENSE file in the project root for
+details.
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+"""
+
+import os
+import importlib
+
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives import unchanged_required
+from docutils import nodes
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+from configsuite.extension.ext_nodes import (
+    visit_children,
+    visit_container,
+    visit_content,
+    visit_header,
+    depart_children,
+    depart_container,
+    depart_content,
+    depart_header,
+    cs_children,
+    cs_container,
+    cs_content,
+    cs_header,
+)
+
+BASIC_EXAMPLE = {
+    "string": "default",
+    "integer": "0",
+    "number": "0.0",
+    "bool": "True",
+    "date": "01.01.1999",
+    "datetime": "01.01.1999",
+}
+
+
+def scb_static_path(app):
+    static_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "_static"))
+    app.config.html_static_path.append(static_path)
+
+
+# we are assuming several keywords.
+# I think we should provide proper user feedback if these are not included.
+def generate(schema, parent_section, parent_title=""):
+    title = schema.get(MK.Description, "default_title")[:10]
+    title = title.lower()
+    title = title.replace(" ", "_")
+    example = None
+
+    section = cs_container(
+        classes=["cs_container"], ids=["cs_{}_container".format(title)]
+    )
+
+    section += cs_header(classes=["cs_header"], text=parent_title)
+
+    section_content = cs_content(
+        classes=["cs_content"], ids=["cs_{}_content".format(title)]
+    )
+
+    section_content += nodes.paragraph(text=schema.get(MK.Description, ""))
+
+    for config_key in [MK.ElementValidators, MK.ContextValidators]:
+        element_nodes = [
+            nodes.paragraph(text=validator.msg)
+            for validator in schema.get(config_key, [])
+        ]
+        section_content.extend(element_nodes)
+
+    for config_key in [MK.LayerTransformation, MK.Transformation]:
+        if config_key in schema:
+            section_content += nodes.paragraph(text=schema.get(config_key).msg)
```

Comment:
Is the `msg` part sufficient? Could we even use anything else?

Reply:
For the user of configsuite, I don't think it's obvious that the `msg` added to the decorators is what will inform the user in the documentation about what is going on. Also, I feel this splits the information about what is going into the documentation quite significantly. All the descriptions are already within the schema dict - and then you would have to go to the function declaration in order to edit documentation regarding transformations/validations..

And those `msg` looks to be one-liners, while transformations and validations, especially context based ones, could probably need more than that

Let's think about it a bit at least..

Reply:
https://github.com/equinor/configsuite/issues/114